sql/upgrade/sl3.0.sql

```diff
@@ -461,12 +474,34 @@ UPDATE :slschema.vendor SET entity_id = (SELECT id FROM entity WHERE 'V-' || ven
 
 UPDATE :slschema.customer SET entity_id = coalesce((SELECT min(id) FROM entity WHERE 'C-' || customernumber = control_code), entity_id);
 
+--TODO: Do we need that for MC?
+INSERT INTO defaults(setting_key,value)
+    SELECT 'curr',array_to_string(array_agg(curr),':')
+    FROM :slschema.curr
+    WHERE rn=1;
+
+INSERT INTO currency(curr,description)
+    SELECT curr,curr
+    FROM  :slschema.curr;
+
+-- Make sure currency table is complete
+SELECT DISTINCT curr, curr
+FROM (
+          SELECT DISTINCT curr FROM :slschema.ar
+    UNION SELECT DISTINCT curr FROM :slschema.ap
+    UNION SELECT DISTINCT curr FROM :slschema.gl
+) xx
+WHERE NOT EXISTS (
+    SELECT 1 FROM :slschema.curr c
+    WHERE c.curr = xx.curr AND curr is not null
+);
```

Comment:
You can simplify the WHERE clause by simply moving the first select into the FROM clause UNION too. That way, it'll simply be one of the factors.

Reply:
Fixed